symfony/src/Entity/DeceasedReport.php

```diff
@@ -0,0 +1,315 @@
+    }
+
+    public function getDenialReasonOtherDescription()
+    {
+        return $this->denialReasonOtherDescription;
+    }
+
+    public function setDenialReasonOtherDescription($denialReasonOtherDescription)
+    {
+        $this->denialReasonOtherDescription = $denialReasonOtherDescription;
+        return $this;
+    }
+
+    /* Model Methods */
+
+    public function getReportmechanismDisplay()
+    {
+        if (isset(self::MECHANISMS[$this->reportMechanism])) {
+            return self::MECHANISMS[$this->reportMechanism];
+        }
+        return $this->reportMechanism;
+    }
+
+    public function getReportStatusDisplay()
+    {
+        if (isset(self::STATUSES[$this->reportStatus])) {
+            return self::STATUSES[$this->reportStatus];
+        }
+        return $this->reportStatus;
+    }
+
+    public function getNextOfKinRelationshipDisplay()
+    {
+        if (isset(self::NK_RELATIONSHIPS[$this->nextOfKinRelationship])) {
+            return self::NK_RELATIONSHIPS[$this->nextOfKinRelationship];
+        }
+        return $this->nextOfKinRelationship;
+    }
+
+    public function loadFromFhirObservation($report)
+    {
+        $this->setId($report->identifier[0]->value);
+        $this->setParticipantId($report->subject->reference);
+        if (property_exists($report, 'effectiveDateTime') && $report->effectiveDateTime) {
+            $this->setDateOfDeath(new \DateTime($report->effectiveDateTime));
+        }
+        if (property_exists($report, 'valueString') && $report->valueString) {
+            $this->setCauseOfDeath($report->valueString);
+        }
+        $this->setReportMechanism($report->encounter->reference);
+        if (property_exists($report->encounter, 'display')) {
+            $this->setReportMechanismOtherDescription($report->encounter->display);
+        }
+        $this->setReportStatus($report->status);
+        foreach ($report->performer as $performer) {
+            switch ($performer->extension[0]->url) {
+                case 'https://www.pmi-ops.org/observation/authored':
+                    $this->setSubmittedBy($performer->reference);
+                    $this->setSubmittedOn(new \DateTime($performer->extension[0]->valueDateTime));
+                    break;
+                case 'https://www.pmi-ops.org/observation/reviewed':
+                    $this->setReviewedBy($performer->reference);
+                    $this->setReviewedOn(new \DateTime($performer->extension[0]->valueDateTime));
+                    break;
+            }
+
+        }
+
+        if (property_exists($report->extension[0], 'valueHumanName') && $report->extension[0]->valueHumanName) {
+            $this->setNextOfKinName($report->extension[0]->valueHumanName->text);
+            $this->setNextOfKinRelationship($report->extension[0]->valueHumanName->extension[0]->valueCode);
+            $this->setNextOfKinTelephoneNumber($report->extension[0]->valueHumanName->extension[1]->valueString);
```

Comment:
Similarly, I got an error as the index 1 is not defined `valueHumanName->extension[1]`. I think we need make sure that these indexes are defined.

Reply:
Fixed in 4409e11